### PR TITLE
Proceed with removal if user not found

### PR DESCRIPTION
**Pull request by rajivnathan**
Fix for bug https://issues.redhat.com/browse/CRT-891

Update the Che user deletion logic such that if the UserExists check comes back with a 404 then we won't continue further to call the DELETE API, we'll assume the Che data has been removed and continue with deactivation/removal.

It also removes the logic to remove the setting of the special condition that was used for caching the cheUserID value between attempts. We no longer need to cache the cheUserID because if we can't get it due to a 404 then it now means the user is removed, if cannot get it due to another response code then it will be an error condition set on the useraccount.

### CODE REVIEW

**Review by alexeykazakov — APPROVED** [review]
Nice cleanup!

Let's delete not used ConditionType and Reasons from API **after** we merge this PR and confirm everything works as expected.

**Review by MatousJobanek — APPROVED** [review]
Nice :+1: :1st_place_medal: 
in the next PR, we could make the deletion of the Che data as the last action of the finalizer. With this, it will be easier to unblock any potential problematic deletions by manual removing the finalized from the resource.

**Comment by rajivnathan** [comment]
> Nice cleanup!
> 
> Let's delete not used ConditionType and Reasons from API **after** we merge this PR and confirm everything works as expected.

PR up here: https://github.com/codeready-toolchain/api/pull/208

**Pull request merged**
